RedHatInsights/insights-inventory-frontend
# chore: Set up cypress and component testing

This installs the cypress and its dependencies, creates default configuration files, adds npm commands and creates an  example spec to verify that components are built correctly using cypress webpack config.

## How to test

1. You should be able to run `npm run test:ct` and `npm run test:openct`. Both commands must show the only example component test with the spinner component.
2. Running both `npm run test` and `npm run test:ct` must generate `codecov-jest` and `codecov-cypress` directories accordingly. Running `npm run coverage` must generate the final coverage report and attempt to upload it to a server.

- [x] Install cypress and dependency libraries,
- [x] Create default configuration files,
- [x] Add npm scripts,
- [x] Write an example componen test,
- [x] Integrate cypress to the code coverage.
